packages/charts/src/utils.ts

```diff
@@ -210,3 +218,42 @@ export const generateTicks = ({
 
   return ticks;
 };
+
+export const calculateColorScale = (
+  minValue: number,
+  maxValue: number,
+  colorsArray?: string[],
+  colorMode: ColorModeType = 'shades'
+) => {
+  const colors =
+    colorsArray && colorsArray.length
+      ? colorsArray.length > 1
+        ? colorsArray
+        : ['white', colorsArray[0]]
+      : ['white', theme.colors[0]];
+  // steps
+  if (colorMode === 'steps' && colorsArray.length > 1) {
```

Comment:
`colors.length` is invoked here multiple times - please define `colorsCount` variable to avoid prototype delegation.